beta/components/theme-switcher/index.js

```diff
@@ -84,6 +84,16 @@ class ThemeSwitcher extends Component {
   load (el) {
     this.local.auto = localStorage !== null && !!localStorage.getItem('color-scheme-auto')
     this.rerender()
+    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Semicolon should be removed

Reply:
I would also check if matchMedia is available

Reply:
```suggestion
    if (
      window !== undefined &&
      window.matchMedia('(prefers-color-scheme: dark)') !== undefined &&
      window.matchMedia('(prefers-color-scheme: dark)').matches !== undefined
      ) {
        const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches
```

Would adding the above checks be suitable for checking if `matchMedia` is available? The above suggestion would of course also need the closing bracket for the new if block below `this.local.machine.emit('theme:toggle')`.

Logging these three checks in the console appear to consistently yield `true` in my testing:
```js
console.log(window !== undefined)
console.log(window.matchMedia('(prefers-color-scheme: dark)') !== undefined)
console.log(window.matchMedia('(prefers-color-scheme: dark)').matches !== undefined)
```

Reply:
I've addressed your suggestions here: https://github.com/resonatecoop/stream/pull/128/commits/3bf0f13687122e596000945bcb921239bfdb8839.

Let me know if some of these `!== undefined` checks I've added in the `if` block are excessive, if so, I'm happy to remove them.

It seems that there may be [some cases](https://stackoverflow.com/questions/39830580/jest-test-fails-typeerror-window-matchmedia-is-not-a-function) where the `window` object doesn't have the `matchMedia` property, though I haven't encountered this yet in testing this repository.